haskell/haskell.bzl

```diff
@@ -1205,12 +1211,37 @@ def haskell_binary_impl(ctx: AnalysisContext) -> list[Provider]:
         )
         run.hidden(symlink_dir)
 
+    sub_targets = {}
+    sub_targets.update(_haskell_module_sub_targets(
+        compiled = compiled,
+        link_style = link_style,
+        enable_profiling = enable_profiling,
+    ))
+
     providers = [
-        DefaultInfo(default_output = output),
+        DefaultInfo(
+            default_output = output,
+            sub_targets = sub_targets,
```

Comment:
For some reason this does not seem to work on `haskell_binary` targets, yet. The following lists no sub-targets:
```
$ buck2 audit subtargets //backend/one-off-task:one-off-task
```
Per print debugging the Starlark code this should list per-module objects and interfaces sub-targets.

Reply:
Nevermind, I just missed the `-hs` suffix in the target name.
```
$ buck2 audit subtargets //backend/one-off-task:one-off-task-hs
```
Works as expected.